kirill_tolstobrov/main.cpp

```diff
@@ -143,49 +207,136 @@ class Graph {
 
   // connections_map: vertex1 -> edge -> vertex2
   std::map<VertexId, std::vector<EdgeId>> connections_map_;
+  std::vector<std::pair<int, int>> depths_map;
```

Comment:
1) `Naming Convention`
2) А не проще будет
```suggestion
  std::vector<std::vector<VertexId>> depths_map_;
```